### ISSUE TITLE
URLError: <urlopen error no host given>

### ISSUE THREAD

**Issue by Vishu92** [pytube/pytube#122]
I am trying to run the package on windows 7, python 3.5 but getting this error
```
from` bs4 import BeautifulSoup`
import requests
from pytube import YouTube

print("Enter the text search")
query=input()
url = "https://www.youtube.com/results?search_query=" + query
response = requests.get(url)
html = response.text
soup = BeautifulSoup(html, 'lxml')

links = []

for vid in soup.findAll(attrs={'class':'yt-uix-tile-link'}):
    links.append('https://www.youtube.com' + vid['href'])

links.pop(0)

for a in links:
	yt = YouTube(a)
	print(yt.get_videos())
```
But every time I am getting this error:

> Traceback (most recent call last):
>   File "youtube.py", line 20, in <module>
>     yt = YouTube(a)
>   File "C:\Users\Vishu\Anaconda3\lib\site-packages\pytube\api.py", line 66, in _
> _init__
>     self.from_url(url)
>   File "C:\Users\Vishu\Anaconda3\lib\site-packages\pytube\api.py", line 189, in
> from_url
>     signature = self._get_cipher(stream_map["s"][i], js_url)
>   File "C:\Users\Vishu\Anaconda3\lib\site-packages\pytube\api.py", line 363, in
> _get_cipher
>     response = urlopen(url)
>   File "C:\Users\Vishu\Anaconda3\lib\urllib\request.py", line 163, in urlopen
>     return opener.open(url, data, timeout)
>   File "C:\Users\Vishu\Anaconda3\lib\urllib\request.py", line 464, in open
>     req = meth(req)
>   File "C:\Users\Vishu\Anaconda3\lib\urllib\request.py", line 1176, in do_reques
> t_
>     raise URLError('no host given')
> urllib.error.URLError: <urlopen error no host given>
> 

```



```

**Comment by ghost** [comment]
Same issue with Windows 10 and Python 3.1. However, I wanted to add that I tried with the Windows 10 bash shell, and it worked. Moreover, the issue does not seem to be tied with particular URLs : when downloading more than a hundred videos a few times, I noticed that the videos that threw this error were not always the same.

**Comment by palashgoel7** [comment]
Tested on Linux Mint 18 with python 3.5.2 and I am not able to hit the issue. @Vishu92 can you print a before sending it to the api.

**Comment by Vishu92** [comment]
@palashgoel7 Issue has been resolved. Now it is working.

**Comment by Vishu92** [comment]
@palashgoel7 I don't know what is wrong but it just worked then it is again showing same error

**Comment by palashgoel7** [comment]
YouTube doesn't allow downloading some videos. For such videos, it does not supply the signature and we have to look elsewhere to get the signature. That is what _get_cipher function is for. 
The issue has already been addressed by @rotemvilsa in #120. Please download the latest version and see if the problem persists.
